src/PollingBlockTracker.test.ts

```diff
@@ -185,6 +185,89 @@ describe('PollingBlockTracker', () => {
       );
     });
 
+    it('should not retry failed requests after the block tracker is stopped', async () => {
+      recordCallsToSetTimeout({ numAutomaticCalls: 1 });
+
+      await withPollingBlockTracker(
+        {
+          provider: {
+            stubs: [
+              {
+                methodName: 'eth_blockNumber',
+                error: 'boom',
+              },
+            ],
+          },
+        },
+        async ({ blockTracker }) => {
+          const latestBlockPromise = blockTracker.getLatestBlock();
+
+          expect(blockTracker.isRunning()).toBe(true);
+          await blockTracker.destroy();
+          await expect(latestBlockPromise).rejects.toThrow(
+            'Block tracker destroyed',
+          );
+          expect(blockTracker.isRunning()).toBe(false);
+        },
+      );
+    });
+
+    it('should return the same promise if called multiple times', async () => {
+      await withPollingBlockTracker(async ({ blockTracker }) => {
+        const promiseToGetLatestBlock1 = blockTracker.getLatestBlock();
+        const promiseToGetLatestBlock2 = blockTracker.getLatestBlock();
+
+        expect(promiseToGetLatestBlock1).toStrictEqual(
+          promiseToGetLatestBlock2,
+        );
```

Comment:
Nit: Should this test use `toBe` to indicate that the promises should be the exact same object?

``` suggestion
        expect(promiseToGetLatestBlock1).toBe(promiseToGetLatestBlock2);